### ISSUE TITLE
[bug] Issue loading MNIST dataset

### ISSUE THREAD

**Issue by sedurCode** [prabhuomkar/pytorch-cpp#83]
**Describe the bug**
There are two parts to this bug.
First off I am failing to build because CMAKE is telling me that the MD5sums of the 4 files from the classic source that is referenced are different to those in the CMakeLists file. 

- This can be bypassed by accessing the mnist files [stashed in google by pytorch](https://storage.googleapis.com/torch-datasets/mnist/)

Bypassing that issue, as I reach the part of the example where the dataset is loaded, a std::bad_alloc exception is thrown.

**To Reproduce**
- Load the project directory in VisualStudio 2019
- Open the CMakeLists configuration
- Select the torch basis example project from the build targets
- (build and) run the project

**Expected behavior**
The project would successfully pass through the input pipeline section of the example

**Screenshots**
![image](https://user-images.githubusercontent.com/25512064/123610555-9d8c2200-d7f8-11eb-8e52-c49fcaa3ee0b.png)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Libtorch Version "1.9.0+cu111" 

**Additional context**
I get the same issue using the auto-downloaded version of libtorch

**Comment by mfl28** [comment]
Hi @sedurCode !

There seem to be some problems with the download from the classic MNIST source (http://yann.lecun.com/exdb/mnist/) from time to time. If you get the correct files from another source and place them inside data/mnist you should be OK.

For your second problem: This may be connected to the relative path (`MNIST_data_path`) that is set in the pytorch-basics `main.cpp`. By default VS uses `<project_dir>\out\build\<config_name>` as build root folder. However the relative path for the MNIST data is set up with the assumption that the build root is at `<project_dir>\build`. To fix this, you can either change the relative path in the file or set the build root directory to the absolute path `path\to\pytorch-cpp\build` in Visual Studio (drop-down menu for the configuration -> Manage Configurations... -> Build root).

**Comment by sedurCode** [comment]
Thanks @mfl28 I changed the file to point at the absolute path with escaped slashes, but issue was not resolved. 
To better track the bug I swapped over to the debug build of the same version of libtorch, and now I am getting past the issue. 
Not sure if its worth further chasing, as I suspect it was finger trouble on my part in the cmakelists settings json file for the project.
Thanks for responding to me very quickly!
